Cannot build on CentOS6: C++ compiler is needed

**Operating system**: CentOS6/RHEL6
**wxPython version**: 4.0.1/master
**Python version**: 2.7.14
**Description of the problem**:
Cannot compile on RHEL6, or a CentOS docker image of the same environment. Encountered on a production RHEL6 system via pip, reproduced on a CentOS6 `Dockerfile`:
```Dockerfile
FROM centos:6

RUN yum install -y gcc gcc-c++ make zlib-devel openssl-devel git

# WX dependencies, not required to demonstrate this problem
# RUN yum install -y  freeglut freeglut-devel gstreamer-plugins-base-devel gtk2-devel libjpeg-devel libnotify-devel libSM-devel libtiff-devel libXtst-devel SDL-devel webkitgtk-devel

# Install python 2.7
RUN mkdir /tmp/python && \
    cd /tmp/python && \
    curl -L https://www.python.org/ftp/python/2.7.14/Python-2.7.14.tgz | tar -xz --strip-components=1 && \
    ./configure --prefix=/opt/python && \
    make install
ENV PATH="/opt/python/bin:${PATH}"

# Get the source and build
RUN cd /tmp && git clone --recursive https://github.com/wxWidgets/Phoenix.git
RUN cd /tmp/Phoenix && python build.py --gtk2 build
```

Get the error during configuration of wx:

```
...
checking for g++... g++
checking whether we are using the GNU C++ compiler... no
checking whether g++ accepts -g... yes
checking whether we are using the Intel C++ compiler... no
checking whether we are using the Sun C++ compiler... no
configure: error: C++ compiler is needed to build wxWidgets
Traceback (most recent call last):
  File "build.py", line 1295, in cmd_build_wx
    wxbuild.main(wxDir(), build_options)
  File "/tmp/Phoenix/buildtools/build_wxwidgets.py", line 374, in main
    "Error running configure")
  File "/tmp/Phoenix/buildtools/build_wxwidgets.py", line 85, in exitIfError
    raise builder.BuildError(msg)
BuildError
```

However, calling the configuration that the build prints out directly:
```
$ /tmp/Phoenix/ext/wxWidgets/configure --enable-unicode --with-gtk=2 --with-opengl --enable-sound --enable-graphics_ctx --enable-mediactrl --enable-display --enable-geometry --enable-debug_flag --enable-optimise --disable-debugreport --enable-uiactionsim --with-sdl
```
**works**, so this appears to be a problem in the way Phoenix calls the wx builder, not a problem of wx directly?

--- a/build.py
+++ b/build.py
@@ -783,7 +783,11 @@ def checkCompiler(quiet=False):
     # causes problems with other non-Windows, non-Darwin compilers then
     # we'll need to make this a little smarter about what flag (if any)
     # needs to be used.
-    if not isWindows and not isDarwin:
+    #
+    # NOTE 2: SIP chenged its output such that this doesn't appear to be
+    # needed anymore, but we'll leave the code in place to make it easy to
+    # turn it back on again if/when needed.
+    if False and not isWindows and not isDarwin:
         stdflag = '-std=c++11'
         curflags = os.environ.get('CXXFLAGS', '')
         if stdflag not in curflags:

